bodo/hiframes/pd_multi_index_ext.py

```diff
@@ -28,13 +29,24 @@
     is_overload_none,
 )
 
+IndexNameType: pt.TypeAlias = (
```

Comment:
Again, for typechecking purposes, so later devs know what the names can be

Reply:
Technically, Index name can be any object in Pandas. Bodo's support seems generic too (see tuple example below). But it's fine to restrict it for Parquet write and throw errors there.

```python
In [7]: A = pd.Index([1, 2], name=(3,4))

In [8]: bodo.jit(spawn=False)(lambda A: A.name)(A)
/Users/ehsan/dev/Bodo/bodo/transforms/distributed_analysis.py:305: BodoWarning: No parallelism found for function '<lambda>'. Distributed diagnostics:
Setting distribution of variable 'A' to REP: Distributed analysis replicated argument 'A' (variable 'A'). Set distributed flag for 'A' if distributed partitions are passed (e.g. @bodo.jit(distributed=['A'])).

File "<ipython-input-8-8a06ad0bec01>", line 1:
bodo.jit(spawn=False)(lambda A: A.name)(A)
^
  warnings.warn(
Out[8]: (3, 4)
```

Reply:
Yes I believe it is restricted to string names right now because thats all Parquet can support. But this is just for typing in Bodo

Reply:
Added